main.tf

```diff
@@ -1,3 +1,73 @@
-resource "random_pet" "template" {
-  length = var.length
+locals {
+  memberships_map = merge([
+    for team_name, team in var.teams : {
+      for member in team.members : "${team_name}:${member.username}" => {
+        team_id  = github_team.default[team_name].id
+        username = member.username
+        role     = member.role
+      }
+    }
+  ]...)
+
+  teams_with_delegation = {
+    for team_name, team in var.teams :
+    team_name => team
+    if team.review_request_delegation != null
+  }
+
+  repository_collaborators = merge([
+    for repo_name, collaborators in var.repository_collaborators : {
+      for collab in collaborators : "${repo_name}:${collab.username}" => {
+        repository                  = repo_name
+        username                    = collab.username
+        permission                  = collab.permission
+        permission_diff_suppression = collab.permission_diff_suppression
+      }
+    }
+  ]...)
+}
+
+resource "github_team" "default" {
+  for_each = var.teams
+
+  name                      = each.value.name
+  description               = each.value.description
+  privacy                   = each.value.privacy
+  parent_team_id            = each.value.parent_team_id
+  ldap_dn                   = each.value.ldap_dn
+  create_default_maintainer = each.value.create_default_maintainer
+}
+
+resource "github_team_membership" "default" {
```

Comment:
Just so I'm clear: Will this invite a new team member to an organization if they're not currently in the org? I think we should make sure that is clear in the README as well -- important info.